### ISSUE TITLE
`style_active_file()` struggles with some code chunk labels containing `-`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/styler#838]
[This](https://bookdown.org/yihui/rmarkdown/r-code.html) states that dashes are allowed in code chunk labels in markdown.

But if I run `styler:::style_active_file()` (via RStudio addin) on the following `.Rmd` file:

````
---
output: rmarkdown::html_document
---

```{r label-1}
x = 5
```

```{r label-in}
x = 5
```
````

it produces the following error:

```r
Using style transformers `styler::tidyverse_style()`
Error: <text>:1:10: unexpected 'in'
1: c( label-in
             ^
```

This is very confusing because it doesn't error with `label-1`, but does with `label-in`.

Trace for the error:

```r
> rlang::last_trace()
<error/rlang_error>
<text>:1:10: unexpected 'in'
1: c( label-in
             ^
Backtrace:
     x
  1. +-styler:::style_active_file()
  2. | \-styler:::transform_mixed(context$contents, transformer, filetype = "Rmd")
  3. |   \-styler:::separate_chunks(lines, filetype)
  4. |     \-styler:::identify_raw_chunks(lines, filetype = filetype)
  5. |       +-`%>%`(...)
  6. |       \-purrr::map2(...)
  7. |         \-styler:::.f(.x[[i]], .y[[i]], ...)
  8. |           \-styler:::get_parse_data(paste0("c(", header, ")"))
  9. |             \-styler:::parse_safely(text, keep.source = TRUE)
 10. +-purrr::transpose(.)
 11. \-purrr::compact(.)
 12.   \-purrr::discard(.x, function(x) is_empty(.f(x)))
 13.     \-purrr:::probe(.x, .p, ...)
 14.       \-purrr::map_lgl(.x, .p, ...)
```

**Comment by lorenzwalthert** [comment]
Are you on the devel version? I think this was fixed in #832.

**Comment by IndrajeetPatil** [comment]
Oops, my bad. For once, I was not working with the development version 😅 

I can confirm that the PR has already fixed this issue!
